### ISSUE TITLE
feat(ai): Implementar classificação de solo com TensorFlow Lite

### ISSUE THREAD

**Issue by LukeSantossz** [LukeSantossz/visiosoil-app#4]
## Descrição

Integrar modelo de IA para classificação automática de textura do solo baseado na imagem capturada. Core feature da Fase 2 do projeto.

## Tipo

- [ ] Bug
- [x] Feature
- [ ] Refatoração
- [ ] Documentação
- [ ] Outro

## Comportamento Esperado

Após captura da foto, o app executa inferência on-device com TFLite e exibe a classe de textura + score de confiança.

## Comportamento Atual

PR #16 em andamento. InferenceService implementado com execução em isolate separado. Modelo placeholder pendente de treinamento.

## Contexto

- **Dispositivo:** Android / iOS
- **Versão Flutter:** 3.38.5
- **Branch:** dev

## Task Relacionada

TASK-001

## Critérios de Aceite

- [x] Integrar pacote `tflite_flutter`
- [x] Criar `InferenceService` com execução em isolate
- [x] Processar imagem e retornar classificação (12 classes USDA)
- [x] Exibir resultado na tela de captura e detalhes
- [x] Armazenar `textureClass` e `confidenceScore` no `SoilRecord`
- [ ] Modelo de produção treinado e embarcado

**Comment by LukeSantossz** [comment]
## Atualização — Pipeline de Classificação Definido

### Modelo Treinado
- **Algoritmo:** Logistic Regression (scikit-learn 1.5.2) via Orange Data Mining
- **Arquivo:** `smartsoil.pkcls` (56 MB)
- **Features:** 1000 (geradas por Image Embedding — SqueezeNet 1.1)
- **Classes ativas:** 3 — Arenosa (0), Argilosa (1), Média (2)
- **Classes futuras:** 5 — Arenosa, Média, Argilosa, Muito Argilosa, Siltosa

### Pipeline On-Device
```
Imagem (224×224 RGB)
  → Normalização ImageNet (mean/std)
  → SqueezeNet 1.1 TFLite (feature extractor, sem softmax)
  → 1000 raw logits
  → Logistic Regression (Dart puro: X·Wᵀ + b → softmax)
  → Classe + Confiança
```

### Plano de Implementação
1. **Extrair pesos LR** do `.pkcls` → `assets/models/lr_weights.json` (coef 3×1000 + intercept)
2. **Obter SqueezeNet 1.1 TFLite** sem softmax (PyTorch → ONNX → TFLite via onnx2tf, ou download Qualcomm AI Hub)
3. **Reescrever `inference_service.dart`** — pipeline dois estágios: SqueezeNet TFLite + LR em Dart
4. **Atualizar preprocessing** — ImageNet normalization (mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225])
5. **Build APK** para teste em dispositivo real

### Status Atual
- [x] Integrar pacote `tflite_flutter`
- [x] `InferenceService` com execução em Isolate (non-blocking)
- [x] Exibir resultado na tela de captura e detalhes
- [x] Armazenar `textureClass` e `confidenceScore` no `SoilRecord`
- [ ] Converter/obter SqueezeNet 1.1 TFLite (sem softmax)
- [ ] Extrair pesos LR para JSON
- [ ] Implementar pipeline dois estágios (SqueezeNet + LR Dart)
- [ ] Ajustar preprocessing para ImageNet normalization
- [ ] Testar classificação em dispositivo real
